src/participant/participant.ts

```diff
@@ -9,39 +9,40 @@ import EXTENSION_COMMANDS from '../commands';
 import type { StorageController } from '../storage';
 import { StorageVariables } from '../storage';
 import { GenericPrompt } from './prompts/generic';
-import { CHAT_PARTICIPANT_ID } from './constants';
+import {
+  AskToConnectChatResult,
+  CHAT_PARTICIPANT_ID,
+  EmptyRequestChatResult,
+  NamespaceRequestChatResult,
+  RunnableChatResult,
+} from './constants';
 import { QueryPrompt } from './prompts/query';
 import { COL_NAME_ID, DB_NAME_ID, NamespacePrompt } from './prompts/namespace';
 import { SchemaFormatter } from './schema';
 import { getSimplifiedSampleDocuments } from './sampleDocuments';
 import { getCopilotModel } from './model';
+import { createMarkdownLink } from './markdown';
+import {
+  collectionNameInCommandResponseKey,
+  databaseNameInCommandResponseKey,
+  getLatestDatabaseAndCollectionFromChatHistory,
+} from './prompts/history';
 
 const log = createLogger('participant');
 
-export enum QUERY_GENERATION_STATE {
-  DEFAULT = 'DEFAULT',
-  ASK_TO_CONNECT = 'ASK_TO_CONNECT',
-  ASK_FOR_DATABASE_NAME = 'ASK_FOR_DATABASE_NAME',
-  ASK_FOR_COLLECTION_NAME = 'ASK_FOR_COLLECTION_NAME',
-  CHANGE_DATABASE_NAME = 'CHANGE_DATABASE_NAME',
```

Comment:
I think we lost the CHANGE_DATABASE_NAME state with refactoring. When a user has generated a query and then clicked on a different database name, we previously displayed the collections list again, so they can choose a new collection name. Now we just use the new database name with the old collection name, which most likely won't exist in the new namespace.

Reply:
Updated to return when we find the first database name in the history, so if the collection name isn't there by then we don't have it so it'll prompt. Good catch!